scm/driver/harness/pr.go

```diff
@@ -99,44 +99,38 @@ func (s *pullService) Close(context.Context, string, int) (*scm.Response, error)
 // native data structures
 type (
 	pr struct {
-		Author struct {
-			Created     int    `json:"created"`
-			DisplayName string `json:"display_name"`
-			Email       string `json:"email"`
-			ID          int    `json:"id"`
-			Type        string `json:"type"`
-			UID         string `json:"uid"`
-			Updated     int    `json:"updated"`
-		} `json:"author"`
-		Created       int    `json:"created"`
-		Description   string `json:"description"`
-		Edited        int    `json:"edited"`
-		IsDraft       bool   `json:"is_draft"`
-		MergeBaseSha  string `json:"merge_base_sha"`
-		MergeHeadSha  string `json:"merge_head_sha"`
-		MergeStrategy string `json:"merge_strategy"`
-		Merged        int    `json:"merged"`
-		Merger        struct {
-			Created     int    `json:"created"`
-			DisplayName string `json:"display_name"`
-			Email       string `json:"email"`
-			ID          int    `json:"id"`
-			Type        string `json:"type"`
-			UID         string `json:"uid"`
-			Updated     int    `json:"updated"`
-		} `json:"merger"`
-		Number       int    `json:"number"`
+		Author      principal `json:"author"`
+		Created     int64     `json:"created"`
+		Description string    `json:"description"`
+		Edited      int64     `json:"edited"`
+		IsDraft     bool      `json:"is_draft"`
+
+		MergeTargetSHA   string    `json:"merge_target_sha"`
+		MergeBaseSha     string    `json:"merge_base_sha"`
+		Merged           int       `json:"merged"`
+		MergeMethod      string    `json:"merge_method"`
+		MergeSHA         string    `json:"merge_sha"`
+		MergeCheckStatus string    `json:"merge_check_status"`
+		MergeConflicts   []string  `json:"merge_conflicts,omitempty"`
+		Merger           principal `json:"merger"`
+
+		Number int `json:"number"`
+
 		SourceBranch string `json:"source_branch"`
 		SourceRepoID int    `json:"source_repo_id"`
```

Comment:
int64

Reply:
cant make this change as base struct only supports int

Reply:
I think we should still have int64 here to stay true - and then when converting it we cast it.
Otherwise it's very implicit and people might not notice